### PR TITLE
Use EnumReflectionProperty class from persistence package 

### PR DESCRIPTION
**Pull request by greg0ire**
It is meant to remove duplication between the ORM and the ODM.

### CODE REVIEW

**Comment by greg0ire** [comment]
Just understood the first error: we need to implement `getAttributes()` in `EnumReflectionProperty`.

**Comment by greg0ire** [comment]
The second failure is on this line: https://github.com/doctrine/orm/blob/b8d0a85017374165d8f8c0855e9eb18806b25d32/src/Mapping/ReflectionEmbeddedProperty.php#L33

@derrabus not sure what we can do about this one: these properties are readonly so [we cannot set them in the constructor](https://3v4l.org/SKaZB).

**Comment by stof** [comment]
Well, if EnumReflectionProperty does not properly populate the parent properties by design, you should change the ReflectionEmbeddedProperty to use the getters instead, making sure that EnumReflectionProperty forwards the getters.

But maybe the class in doctrine/persistence should be changed to actually initialize the parent class as is done in the ORM right now.

**Comment by greg0ire** [comment]
Maybe yes, as proposed in https://github.com/doctrine/persistence/pull/342 :thinking:

**Comment by greg0ire** [comment]
@stof thanks for the help, it does work with the getters 👍 I'll send an alternative to https://github.com/doctrine/persistence/pull/342 and let @derrabus decide

File: UPGRADE.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Upgrade to 3.1
 
+## Deprecate `Doctrine\ORM\Mapping\ReflectionEnumProperty`
+
+This class is deprecated and will be removed in 4.0.
+Instead, use `Doctrine\Persistence\ReflectionEnumProperty` from
```
Comment (stetodd) [inline UPGRADE.md]: Should this read:
```suggestion
Instead, use `Doctrine\Persistence\Reflection\EnumReflectionProperty` from
```

**Review by derrabus — APPROVED** [review]

**Pull request merged**